### PR TITLE
Use the coveralls Github action

### PR DESCRIPTION
**Pull request by bmerry**
Also use it with parallel flags, which will hopefully stop the MacOS coverage builds from failing.

### CODE REVIEW

**Review by amishatishpatel — COMMENTED** [review]
Apologies for the delay, I had to read up on GitHub Actions syntax/keywords. Looks fantastic, as always. I have to ask before explicit Approval (genuinely, because I haven't dabbled in the Actions before)
* How have you confirmed this change works?

**Comment by bmerry** [comment]
> Apologies for the delay, I had to read up on GitHub Actions syntax/keywords. Looks fantastic, as always. I have to ask before explicit Approval (genuinely, because I haven't dabbled in the Actions before)
> 
>     * How have you confirmed this change works?

Yep: https://coveralls.io/builds/70463282

**Review by amishatishpatel — APPROVED** [review]

**Pull request merged**
